File: 3024/type-of-triangle.cpp
// 三角形分成:正三角形equilateral、等腰三角isosceles, 其他scalene 
class Solution {
public:
    string triangleType(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        if(nums[0]+nums[1]<=nums[2]) return "none";
        else if(nums[0]==nums[1] && nums[1]==nums[2]) return "equilateral";
        else if(nums[0]==nums[1] || nums[1]==nums[2]) return "isosceles";
        else if(nums[0]+nums[1]>nums[2]) return "scalene";
        else return "none";
    }
};

File: 3024/type-of-triangle.py
# LeetCode 3024. Type of Triangle
# 正三角形、等腰三角形、任意三角形、無法組成三角形
class Solution:
    def triangleType(self, nums: List[int]) -> str:
        a, b, c = sorted(nums)  # 先把3個邊「小到大」排好
        if a+b<=c: return "none"  # 無法組成三角形
        if a==b and b==c: return "equilateral"  # 正三角形
        if a==b or b==c: return "isosceles"  # 等腰三角形
        return "scalene"  # 任意三角形
